src/app/venter/complaints-home/complaints-home.component.ts

```diff
@@ -0,0 +1,150 @@
+import { IComplaint } from '../../interfaces';
+import { DataService } from '../../data.service';
+import { API } from '../../../api';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-complaints-home',
+  templateUrl: './complaints-home.component.html',
+  styleUrls: ['./complaints-home.component.css'],
+})
+export class ComplaintsHomeComponent implements OnInit {
+
+  public complaints = {} as IComplaint[];
+  public myComplaints = {} as IComplaint[];
+  public userid: string;
+  public complaintstatusColor: string[] = [];
+  public myComplaintstatusColor: string[] = [];
+  public reportedTime: string[] = [];
+  public myComplaintsFlag: boolean;
+  public complaintsFlag: boolean;
+
+  constructor(
+    public dataService: DataService,
+    public snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.complaints = [];
+    this.myComplaints = [];
+    /* Get profile if the user is logged in */
+    if (this.dataService.isLoggedIn()) {
+      this.getUser();
+    }
+    /* Set title */
+    this.dataService.setTitle('Complaints & Suggestions');
+    this.refresh();
+
+    /* Get all the complaints from server*/
+    this.dataService.FireGET<IComplaint[]>(API.Complaints).subscribe(result => {
+      this.complaints = result;
+      if (this.complaints.length === 0) {
+        this.complaintsFlag = true;
+      }
+      let i = 0;
+      result.forEach(element => {
+        this.reportedTime[i] = moment(element.report_date).format('MMMM Do YYYY');
+        i = i + 1;
+        if (element.status.toLowerCase() === 'reported') {
+          this.complaintstatusColor.push('red');
+        }
+        else if (element.status.toLowerCase() === 'resolved') {
+          this.complaintstatusColor.push('green');
+        }
+        else if (element.status.toLowerCase() === 'in progress') {
+          this.complaintstatusColor.push('yellow');
+        }
+      });
+    });
+    /* Get the current user complaints from server*/
+    this.dataService.FireGET<IComplaint[]>(API.MyComplaints).subscribe(result => {
+      this.myComplaints = result;
+      if (this.myComplaints.length === 0) {
+        this.myComplaintsFlag = true;
+      }
+      result.forEach(element => {
+        if (element.status.toLowerCase() === 'reported') {
+          this.myComplaintstatusColor.push('red');
+        } else if (element.status.toLowerCase() === 'resolved') {
+          this.myComplaintstatusColor.push('green');
+        } else if (element.status.toLowerCase() === 'in progress') {
+          this.myComplaintstatusColor.push('yellow');
+        }
+      });
+    });
+  }
+
+  getUser() {
+    this.dataService.GetFillCurrentUser().subscribe(user => {
+      this.userid = user.id;
+    });
+  }
+
+  refresh() {
+    /* Get all the complaints from server*/
+    this.dataService.FireGET<IComplaint[]>(API.Complaints).subscribe(result => {
+      this.complaints = result;
+    });
```

Comment:
Same as above, these calls are async, so even if this works during development, doing this is a terrible idea.